src/main/kotlin/dev/arbjerg/ukulele/features/LeaveOnIdleService.kt

```diff
@@ -0,0 +1,67 @@
+package dev.arbjerg.ukulele.features
+
+import org.springframework.stereotype.Service
+import dev.arbjerg.ukulele.config.BotProps
+import net.dv8tion.jda.api.entities.Guild
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.TaskScheduler
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ScheduledFuture
+
+@Service
+class LeaveOnIdleService(private val scheduler: TaskScheduler, private val botProps: BotProps) {
+    private val log: Logger = LoggerFactory.getLogger(LeaveOnIdleService::class.java)
+    private val timers: ConcurrentHashMap<Long, ScheduledFuture<*>> = ConcurrentHashMap()
+
+    /**
+     * Runnable that gets passed to the Spring TaskScheduler. Used to keep track of how many minutes
+     * a player in a guild has spent idling.
+     */
+    class IdleTask(val guild: Guild, private val idleTimeMinutes: Int) : Runnable {
+        private var numberOfMinutesIdleElapsed = -1
```

Comment:
It's a bit odd to me that you're using a counter and not a unix timestamp, but I suppose this works too.

Reply:
On second thought, could you not just schedule the voice leave after `idleTimeMinutes` minutes?